blog-caas/migration.js

```diff
@@ -0,0 +1,241 @@
+import fs from 'fs';
+import { select } from 'unist-util-select';
+import { u } from 'unist-builder';
+import { BulkUpdate, ExcelReporter, loadListData, saveDocument } from '../bulk-update/index.js';
+import { selectBlock, blockToObject } from '../bulk-update/migration-tools/select.js';
+
+const { pathname } = new URL('.', import.meta.url);
+const DRY_RUN = false; // do not save documents
+const SHORT_RUN = true; // first 50 entries for testing
+
+const MAX_CARD_TITLE_LENGTH = 48;
+const WARNING_CARD_DESCRIPTION_LENGTH = 108;
+const MAX_CARD_DESCRIPTION_LENGTH = 178;
+const PRIMARY_TAG = 'caas:content-type/blog';
+
+function loadCaasMappings() {
+  try {
+    const json = fs.readFileSync(`${pathname}caas-mappings.json`, 'utf8');
+    const mappings = JSON.parse(json);
+    const lowercaseMappings = Object.fromEntries(
+      Object.entries(mappings).map(([key, value]) => [key.toLowerCase(), value]),
+    );
+
+    return lowercaseMappings;
+  } catch (error) {
+    console.error('Error loading caas-mappings.json:', error);
+    return {};
+  }
+}
+
+const caasMappings = loadCaasMappings();
+const dateString = ExcelReporter.getDateString();
+
+const skipList = [
+  '/blog/perspectives/a-quick-start-guide-to-web-performance',
+  '/blog/how-to/how-to-accelerate-content-production-with-adobe-experience-manager-headless',
+  '/blog/perspectives/how-hanesbrands-and-adobe-built-one-of-the-fastest-ecommerce-websites-in-the-world',
+  '/blog/the-latest/digital-experiences-with-adobe-experience-manager-sites',
+  '/blog/basics/a-brief-overview-of-headless-cms',
+  '/blog/how-to/personalizing-ecommerce-merchandising-with-ai-in-adobe-commerce',
+  '/blog/how-to/the-usual-suspects-5-configuration-issues-to-maximize-your-peak-sales',
+  '/blog/basics/digital-asset-management',
+  '/blog/how-to/get-started-with-marketo-engage-program-templates',
+  '/blog/the-latest/streamlining-adobe-experience-manager-sites-content-creation-process-unleashing-efficiency-with-adobe-workfront',
+  '/blog/basics/kanban-boards',
+];
+
+const missingTags = {};
+
+const config = {
+  list: ['https://main--bacom-blog--adobecom.hlx.live/blog/query-index.json'],
+  siteUrl: 'https://main--bacom-blog--adobecom.hlx.live',
+  prodSiteUrl: 'https://business.adobe.com',
+  reporter: new ExcelReporter(`${pathname}reports/caas-${dateString}.xlsx`, false),
+  outputDir: `${pathname}output`,
+  mdDir: `${pathname}md`,
+  mdCacheMs: 24 * 60 * 60 * 1000, // 1 day
+  fetchWaitMs: 20,
+};
```

Comment:
Seems like these could be organized a bit better. Either the constants should be at the top of the file or they should be right above where they're used.

Reply:
Agreed, I'll look at cleaning this up.